File: winmob/XfMobile_WM5/Gamepe/MsgQue.cpp
// Msg.cpp: implementation of the CMsg class.
//
//////////////////////////////////////////////////////////////////////
#include "stdafx.h"
#include<windows.h>
#include "MsgQue.h"

// rivital  02 268    
//////////////////////////////////////////////////////////////////////
// Construction/Destruction
//////////////////////////////////////////////////////////////////////

CMsg::CMsg()
{
	m_pData  = NULL;
	m_dwSize = NULL;
}

CMsg::~CMsg()						   
{
	if (m_pData != NULL) delete m_pData;

	m_pData=NULL;
}

BOOL CMsg::bPut( unsigned char * pData, DWORD dwSize)
{
	m_pData = new char [dwSize+1];
	if (m_pData == NULL) return FALSE;
	ZeroMemory(m_pData, dwSize);
	memcpy(m_pData, pData, dwSize);

	m_dwSize = dwSize;


	return TRUE;
}



void CMsg::Get( unsigned char * pData, DWORD * pSize)
{

	memcpy(pData, m_pData, m_dwSize);
	*pSize  = m_dwSize;
	
}

File: ios/Classes/MsgQue.m
//
//  MsgQue.m
//  iXfire
//
//  Created by Moti Joseph on 4/1/09.
//  Copyright 2009 Gamepe. All rights reserved.
//

#import "MsgQue.h"


@implementation MsgQue



- (id)init
{
	

	m_pData  = NULL;
	m_dwSize = 0;
	return self;
}



- (void)dealloc {
	
   	if (m_pData != NULL) free(m_pData);	
	m_pData=NULL;
	[super dealloc];
}





-(void) bGet:(char *) pFrom  Data:(char *) pData Size:(unsigned short*)  pSize   Index:(int *) pIndex Key: (char *) pKey
{
	*pFrom  = m_cFrom;
	memcpy(pData, m_pData, m_dwSize);
	*pSize  = m_dwSize;
	*pIndex = m_iIndex;
	*pKey   = m_cKey;

	
}




-(int)  bPut:(char) cFrom    Data: (char *) pData  Size:(unsigned short) dwSize   Index:(int) iIndex  Key:( char) cKey
{
	
	m_pData =malloc(dwSize );
	
	if (m_pData == NULL) return FALSE;
	
	memset(m_pData,0,dwSize); 


	memcpy(m_pData, pData, dwSize);
	

	m_dwSize = dwSize;
	
	
	m_cFrom  = cFrom;
	m_iIndex = iIndex;
	m_cKey   = cKey;
	return TRUE;
	
}
@end
